### PR TITLE
KOGITO-1901: Manage columns enhancements

### PR DESCRIPTION
**Pull request by Sara4994**
Please enter the commit message for your changes. Lines starting

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by srambach** [comment]
The refresh icon seems to cause the column management modal to appear. I don't think it should, and should probably not be part of the ManageColumns component since it's really independent of this.

File: packages/common/src/components/Molecules/ManageColumns/ManageColumns.tsx [hunk 1]
```diff
@@ -0,0 +1,628 @@
...
+            Save
+          </Button>,
+          <Button key="cancel" variant="secondary" onClick={handleModalToggle}>
+            Cancel
+          </Button>
+        ]}
+      >
+        <DataList
+          aria-label="Table column management"
+          id="table-column-management"
```
Comment (srambach) [inline packages/common/src/components/Molecules/ManageColumns/ManageColumns.tsx:585]: Please add `isCompact` here.

File: packages/common/src/components/Molecules/ManageColumns/ManageColumns.css [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+.kogito-common--Manage-columns__DataList-content .pf-c-data-list__expandable-content-body .pf-c-data-list__item-control{
+    padding: 10px 0 10px 0;   
+}
+
+.kogito-common--Manage-columns__DataList-content .pf-c-data-list__expandable-content-body .pf-c-data-list__item-content
+    .pf-c-data-list__cell{
+    padding: 10px 0 10px 0;
+}
```
Comment (srambach) [inline packages/common/src/components/Molecules/ManageColumns/ManageColumns.css:20]: Please replace entire contents with that below to clean up the spacing and to use Patternfly spacer variables for consistent spacing. We should test with more than one level of expansion to make sure it still works. This relies on adding the classname `kogito-common--manage-columns__data-list` to the overall table, and a couple of corrections to the classnames also in ManageColumns.tsx
```
.kogito-common--manage-columns__data-list .pf-c-data-list__toggle {
    margin-left: calc(-1 * var(--pf-global--spacer--md));
}

.kogito-common--manage-columns__data-list-content:not(:last-of-type) {
    box-shadow: none;   
}

.kogito-common--manage-columns__data-list-content .pf-c-data-list__expandable-content-body{
    padding: 0px;   
}

.kogito-common--manage-columns__data-list-content .pf-c-data-list__expandable-content-body .pf-c-data-list__item-row {
    padding-left: calc( var(--pf-global--spacer--2xl) + var(--pf-global--spacer--lg));
}

.kogito-common--manage-columns__data-list-content .pf-c-data-list__cell {
    padding-left: var(--pf-global--spacer--md);
}
```

File: packages/common/src/components/Molecules/ManageColumns/ManageColumns.tsx [hunk 3]
```diff
@@ -0,0 +1,628 @@
...
+                }
+              })
+              .map((item, _index) => {
+              allSelections.push(item.name + '/' + title + '/' + group.name)
+              return (
+                <DataListContent
+                  aria-label="Primary Content Details"
+                  id="ex-expand1"
+                  isHidden={!expanded.includes(label.replace(/\,/g, ''))}
+                  className="kogito-common--Manage-columns__DataList-content"
```
Comment (srambach) [inline packages/common/src/components/Molecules/ManageColumns/ManageColumns.tsx:246]: Change to `kogito-common--manage-columns__data-list-content`

File: packages/common/src/components/Molecules/ManageColumns/ManageColumns.tsx [hunk 4]
```diff
@@ -0,0 +1,628 @@
...
+                  })
+                  .map((item, _index) => {
+                  allSelections.push(item.name + '/' + group.name)
+                  return (
+                    <DataListContent
+                      aria-label="Primary Content Details"
+                      id="ex-expand1"
+                      isHidden={!expanded.includes(group.name)}
+                      key={_index}
+                      className="kogito-common--Manage-columns__DataList-content"
```
Comment (srambach) [inline packages/common/src/components/Molecules/ManageColumns/ManageColumns.tsx:363]: Change to `kogito-common--manage-columns__data-list-content`

File: packages/common/src/components/Molecules/ManageColumns/ManageColumns.tsx (hunk 1 above)
Comment (srambach) [reply packages/common/src/components/Molecules/ManageColumns/ManageColumns.tsx:585]: Also, add `className="kogito-common--manage-columns__data-list"` to be able to target this list

**Comment by srambach** [comment]
If we have a Select All option, I think that we also need to be able to remove selection from all. I know this isn't in the Patternfly demo, but we have many more columns to select than are shown in the demo. Is it possible either to use the bulk selector as shown here https://www.patternfly.org/v4/documentation/react/demos/bulkselecttable#component-title, or really all that is needed is the checkbox for all as shown here https://www.patternfly.org/v4/documentation/react/demos/filtertabledemo
Adding a header row with the checkbox and "Column name" as the column name (somewhat ironically) would be fine in my opinion.

**Comment by srambach** [comment]
Here is how I've modified it, for reference along with the code comments. 
image

**Review by srambach — CHANGES_REQUESTED** [review]

**Comment by srambach** [comment]
@Sara4994 I think the bulk selector on the column management modal is getting list bullets because it's within a `pf-c-content` block. Try moving it out of the modal description and down into the modal content where the list is. I think that should solve it.

**Comment by srambach** [comment]
On the empty state, we need to change the wording "Select columns from the dropdown to see content" to something more like "Select [Manage Columns](url) to see content."

**Comment by srambach** [comment]
> @Sara4994 I think the bulk selector on the column management modal is getting list bullets because it's within a `pf-c-content` block. Try moving it out of the modal description and down into the modal content where the list is. I think that should solve it.

And then, let's either use a new class on the dropdown so that we don't increase specificity, and use one of our PF spacer variables. 
```
.kogito-common--manage-columns__dropdown {
    margin-bottom: var(--pf-global--spacer--md);
}
```

**Comment by cristianonicolai** [comment]
@Sara4994 should this be closed in favour of https://github.com/kiegroup/kogito-apps/pull/298 ?

**Comment by Sara4994** [comment]
Opened a new one https://github.com/kiegroup/kogito-apps/pull/298. Closing this

**Pull request closed without merging**